### PR TITLE
P2P-Nathan repair validation errors

### PR DESCRIPTION
**Pull request by P2P-Nathan**
Corrected errors that were preventing the validation success.

### CODE REVIEW

**Comment by P2P-Nathan** [comment]
This PR will enable continuous integration testing, results are visible for the test with the inline links or at https://ci.appveyor.com/project/P2P-Nathan/scom-community-catalog/build/1.0.3/tests

File: Pack.tests.template.ps1 [hunk 1]
```diff
@@ -47,7 +46,7 @@ foreach ($packFile in $packFiles) {
         Context 'details.json' {
             
             It "details.json is valid json" {
-                { $contents | ConvertFrom-Json } | Should Not Throw 
+                {$script:json = $contents | ConvertFrom-Json } | Should Not Throw
```
Comment (shadeon) [inline Pack.tests.template.ps1]: This makes the tests order dependant (other tests will fail if this one isn't run first).  The original call probably needs a try/catch wrapping it.

File: Pack.tests.template.ps1 [hunk 2]
```diff
@@ -33,18 +32,21 @@ foreach ($packFile in $packFiles) {
 
             It "Has a details.json" {
                 $packFile.FullName | Should Exist
+                $packFile.Name | Should BeExactly "details.json"
```
Comment (shadeon) [inline Pack.tests.template.ps1]: In AppVeyor and VSCode only the test name is surfaced - can you either update it to include this restriction or (preferably) create another test?  Good unit testing practise is to only test a single thing with each test.

File: Pack.tests.template.ps1 [hunk 3]
```diff
@@ -33,18 +32,21 @@ foreach ($packFile in $packFiles) {
...
             }
 
             It "Has a Readme.md" {
                 $readme = $packFile.FullName -replace $packFileName, 'README.md'
                 $readme | Should Exist
+                # Get-ChildItem is used as Get-Item returns the same case it is passed.
+                (Get-ChildItem -Path $packFile.DirectoryName -Filter "readme.md").Name  | Should BeExactly "ReadMe.md"
```
Comment (shadeon) [inline Pack.tests.template.ps1]: In AppVeyor and VSCode only the test name is surfaced - can you either update it to include this restriction or (preferably) create another test?  Good unit testing practise is to only test a single thing with each test.

(also, feel free to update my incorrectly cased file name above!)

**Review by shadeon — CHANGES_REQUESTED** [review]
Looks good overall, but some tweaks to the test template are probably necessary to align with best practice.

**Comment by shadeon** [comment]
Might be worth adding this to the top line of the repo readme.md as well:
`Project Badge`

as this will manifest :
Project Badge

**Comment by P2P-Nathan** [comment]
"This makes the tests order dependant (other tests will fail if this one
isn't run first). The original call probably needs a try/catch wrapping it."

Does pester run multi-threaded when performing tested?

**Comment by shadeon** [comment]
Not currently, but it may do in the future.  The other danger is simply the tests being re-ordered.

**Comment by P2P-Nathan** [comment]
Good to know, I'll get that updated, as well as update the tests for the
details.json and ReadMe.md files.
Thank you for the feedback

**Comment by shadeon** [comment]
No problem, just locally tested this and it seemed to do the trick
``` PowerShell
# Even with ErrorAction Silent, this could still throw terminating exceptions
    $json = $null
    try { $json = $contents | ConvertFrom-Json -ErrorAction SilentlyContinue } catch {}
```

File: Pack.tests.template.ps1 (hunk 3 above)
Comment (P2P-Nathan) [reply Pack.tests.template.ps1]: To increase the readability of the output I'll change the test name to "Has a ReadMe.md (case-sensitive)", this should help a user that gets the error even through they have a README.md in the folder.

File: Pack.tests.template.ps1 (hunk 2 above)
Comment (P2P-Nathan) [reply Pack.tests.template.ps1]: To increase the readability of the output I'll change the test name to "Has a details.json (case-sensitive)", this should help a user that gets the error even through they have a details.JSON in the folder.

**Comment by P2P-Nathan** [comment]
The noted changes have been added, and the updated tests have been passed.  Waiting for a community moderator to approve, then we can pull it in.

**Review by shadeon — APPROVED** [review]
Changes look good 👍

**Pull request merged**